SeafShare/ShareViewController.m

```diff
@@ -0,0 +1,363 @@
+@property (nonatomic, strong) SeafConnection *connection;
+@property (nonatomic, strong) SeafDir *directory;
+@property (nonatomic, strong) UIActivityIndicatorView *loadingView;
+@property (nonatomic, strong) UIAlertController *alert;
+@property (weak, nonatomic) IBOutlet UITableView *tableView;
+@property (weak, nonatomic) IBOutlet UILabel *accountLabel;
+@property (weak, nonatomic) IBOutlet UILabel *destinationLabel;
+@property (weak, nonatomic) IBOutlet UIButton *accontButton;
+@property (weak, nonatomic) IBOutlet UIButton *destinationButton;
+@property (weak, nonatomic) IBOutlet UIButton *saveButton;
+
+@end
+
+@implementation ShareViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    if (IsIpad()) {
+        [self setPreferredContentSize:CGSizeMake(480.0f, 540.0f)];
+    }
+    
+    self.navigationItem.leftBarButtonItem = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemCancel target:self action:@selector(cancel:)];
+    self.navigationItem.title = NSLocalizedString(@"Save to Seafile", @"Seafile");
+    
+    self.accountLabel.text = NSLocalizedString(@"Accounts", @"Seafile");
+    self.destinationLabel.text = NSLocalizedString(@"Destination", @"Seafile");
+    self.accontButton.enabled = false;
+    self.destinationButton.enabled = false;
+    [self.saveButton setTitle:NSLocalizedString(@"Save", @"Seafile") forState:UIControlStateNormal];
+    
+    [SeafGlobal.sharedObject loadAccounts];
+    _conns = SeafGlobal.sharedObject.conns;
+    if (_conns.count > 0) {
+        self.connection = _conns.firstObject;
+        NSString *hostAndName = [NSString stringWithFormat:@"%@-%@",_connection.host,_connection.username];
+        [self.accontButton setTitle:hostAndName forState:UIControlStateNormal];
+        [self updateSaveButton];
+    } else {
+        [self showAlert];
+    }
+    
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(selectDirNotification:) name:@"SelectedDirectoryNotif" object:nil];
+    
+    [self setupTableview];
+}
+
+- (void)viewWillAppear:(BOOL)animated {
+    [super viewWillAppear:animated];
+    if (self.ufiles.count == 0) {
+        [self handleInputs];
+    }
+    [self updateSaveButton];
+}
+
+- (void)selectDirNotification:(NSNotification *)notif {
+    self.directory = notif.object;
+    [self updateSaveButton];
+}
+
+- (void)handleInputs {
+    self.group = dispatch_group_create();
+    dispatch_queue_t queue = dispatch_queue_create("com.seafile.share.imagehandle", DISPATCH_QUEUE_CONCURRENT);
+    
+    NSString *tmpdir = [SeafStorage uniqueDirUnder:SeafStorage.sharedObject.tempDir];
+    if (![Utils checkMakeDir:tmpdir]) {
+        Warning("Failed to create temp dir.");
+        return [self alertWithTitle:NSLocalizedString(@"Failed to upload file", @"Seafile") handler:nil];
+    }
+    
+    [self showLoadingView];
+    for (NSExtensionItem *item in self.extensionContext.inputItems) {
```

Comment:
A lot of code shares the same logic as ActionViewController, could we put it into a shared library file?